authors/apps/bookmark/views.py

```diff
@@ -0,0 +1,65 @@
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework.exceptions import APIException
+
+from ..articles.models import Article
+from ..bookmark.models import Bookmark
+from ..authentication.models import User
+
+
+class CreateBookmark(APIView):
+    """This class allows contains views that allow a user to
+    create a bookmark."""
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request, article_title):
+        """The method accepts an article title argument as a string
+        from the URL. Method creates a bookmark object for an
+        existing article. It then creates a relationship between the current
+        user object and the article bookmark."""
+        user = self.request.user
+        username = user.username
+        try:
+            article = Article.objects.get(title=article_title)
+            article_id = article.pk
+        except:
+            return Response({"error": "No article with that title found."},
+                            status=status.HTTP_404_NOT_FOUND)
+        bookmark = Bookmark.objects.filter(article_title=article_title).filter(
```

Comment:
I believe the chained filter can be refactored to  `Bookmark.objects.filter(article_title=article_title,article_id=article_id)`. Are there any particular reasons you chose to do it like this?